test/game_statistics_test.rb

```diff
@@ -1,81 +1,47 @@
-
-    @locations = {
-      games: game_path,
-      teams: team_path,
-      game_teams: game_teams_path
-    }
-
-    @game = Game.new()
-    array_dummy = CSV.read(@locations[:games])
-    @game.create_stat_hash_keys(array_dummy)
+    @game_statistics = GameStatistics.new
   end
 
   def test_it_exists
-    game_statistics = GameStatistics.new(@game.stat_hash)
-    assert_instance_of GameStatistics, game_statistics
-  end
-
-  def test_it_has_a_total_score
-    game_statistics = GameStatistics.new(@game.stat_hash)
-    assert_equal 3, game_statistics.total_goals[2]
-  end
-
-  def test_highest_total_score
-    game_statistics = GameStatistics.new(@game.stat_hash)
-    assert_equal 5, game_statistics.highest_total_score
-  end
-
-  def test_lowest_total_score
-    game_statistics = GameStatistics.new(@game.stat_hash)
-
-    assert_equal 1, game_statistics.lowest_total_score
+    assert_instance_of GameStatistics, @game_statistics
   end
 
-  def test_it_can_determine_percentage_of_home_wins
-    game_statistics = GameStatistics.new(@game.stat_hash)
-    assert_equal 68.42, game_statistics.percentage_home_wins
+  def test_game_statistics_is_an_instance_of_game_data
+    assert_instance_of GameData, @game_statistics.all_games[1]
   end
 
-  def test_count_total_games_by_season
-    game_statistics = GameStatistics.new(@game.stat_hash)
-
-    assert_equal ({"20122013" => 19}), game_statistics.count_games_by_season
+  def test_total_score
+    assert_equal 5, @game_statistics.highest_total_score
+    assert_equal 1, @game_statistics.lowest_total_score
   end
 
-
-  def test_average_goals_by_season
-    game_statistics = GameStatistics.new(@game.stat_hash)
-
-    assert_equal ({"20122013" => 3.68}), game_statistics.average_goals_by_season
+  def test_it_can_determine_percentages
+    @game_statistics.win_data
+    assert_equal 68.42, @game_statistics.percentage_of_home_wins
+    assert_equal 26.32, @game_statistics.percentage_of_visitor_wins
+    assert_equal 5.26, @game_statistics.percentage_of_ties
   end
 
-  def test_average_goals_per_game
-    game_statistics = GameStatistics.new(@game.stat_hash)
-
-    assert_equal 3.68, game_statistics.average_goals_per_game
-
+  def test_it_can_count_games_by_season
+    expected = {20122013=>19}
+    assert_equal expected, @game_statistics.count_of_games_by_season
   end
 
-  def test_it_can_determine_percentage_of_visitor_wins
-    game_statistics = GameStatistics.new(@game.stat_hash)
-    assert_equal 26.32, game_statistics.percentage_visitor_wins
+  def test_it_can_calc_avg_goals_per_game
+    assert_equal 3.68, @game_statistics.average_goals_per_game
   end
 
-  def test_it_can_determine_percentage_ties
-    game_statistics = GameStatistics.new(@game.stat_hash)
-    assert_equal 5.26, game_statistics.percentage_ties
+  def test_it_can_calc_avg_goals_by_season
+    expected = {20122013=>3.68}
+    assert_equal expected, @game_statistics.average_goals_per_season
```

Comment:
I see all your tests

Reply:
I don't think all helper methods have a test. Add this to the refactor list

Reply:
Oh wow! Great catch! Thanks for checking, will add to the todo for Refactoring: GameStatistics